device-identifier/shbdeviceidentifier/utilities/capture_utilities.py

```diff
@@ -0,0 +1,152 @@
+from pathlib import Path
+from typing import List, Union
+
+import pandas as pd
+import pyshark
+from loguru import logger
+from pyshark.capture.capture import Capture
+
+
+def collect_traffic(interface: Union[Path, str] = None, time: int = -1,
+                    output_file: Union[Path, str] = None) -> Union[pyshark.capture.capture.Capture, None]:
+    """
+    The collect_traffic function is used to capture traffic on a specified interface for a specified amount of time.
+    Either the interface_id or the interface name must be specified.
+    If no or non-positive time is specified, it will run indefinitely until the user stops it.
+    It can also write the captured packets to an output file if one is provided.
+
+    Parameters
+    ----------
+        interface: str = None
+            The interface to capture traffic on
+        time: int = 0
+            The time limit for the traffic capture in seconds
+        output_file: Union[Path, str]=None
+            The path to the output file
+
+    Returns
+    -------
+
+        A capture object
+
+    Doc Author
+    ----------
+        TB
+    """
+    if not isinstance(interface, Path) and interface.isdigit():
+        ix = int(interface)
+    else:
+        interface = Path(interface)
+        available_interfaces = [Path(face) for face in pyshark.LiveCapture().interfaces]
+        try:
+            ix = available_interfaces.index(interface)
+        except ValueError:
+            logger.error(f"Interface {interface} not found. Available interfaces are {available_interfaces}.")
+            return None
+
+    interface = pyshark.LiveCapture().interfaces[ix]
+
+    if output_file:
+        output_file = Path(output_file).resolve()
+        logger.info(f"Capturing traffic on interface {interface} and saving to {output_file}.")
+    else:
+        logger.info(f"Capturing traffic on interface {interface}. No output file specified.")
+
+    cap = pyshark.LiveCapture(interface=interface, output_file=output_file)
+
+    if time > 0:
+        cap.sniff(timeout=time)
+    else:
+        logger.error("Continuously capturing traffic is currently not supported.")
+        # TODO: add continuous capture support
+        # logger.info("Sniffing indefinitely... (Ctrl-C to stop)")
+        # gen = cap.sniff_continuously()
+    return cap
+
+
+# noinspection PyPep8Naming
+def convert_Capture_to_DataFrame(cap: Capture) -> pd.DataFrame:
```

Comment:
Why capitalize the words? Is that a naming convention I am unaware of?

Reply:
I felt like this made it clearer that we're talking about a Capture object, not necessarily an arbitrary packet capture file. That being said, I'd be ok with adhering to the PEP convention here as well. 

In general, naming can and should be overhauled for the project as a whole. There is already a ticket for it in the Development Project.